src/library_browser.js

```diff
@@ -1119,6 +1142,10 @@ var LibraryBrowser = {
       return 1; // Return non-zero on failure, can't set timing mode when there is no main loop.
     }
 
+    if (!Browser.mainLoop.timingSet) {
+      {{{ runtimeKeepalivePush(); }}}
+      Browser.mainLoop.timingSet = true;
+    }
```

Comment:
this change looks non-obvious. why unset this here, when it wasn't before? and why does that keep the runtime alive? where is the corresponding pop?

Reply:
This function (emscripten_set_main_loop_timing
) can be used to set the timing of an existing main loop.   If the timing already set we don't want to repeated call runtimeKeepalivePush every time the timing is adjusted.

See `'main loop exiting..'` below in `setMainLoop` for where the loops exits ad calls `Pop`

Reply:
I still find this confusing. Perhaps some comments could help, if I'm missing something.

It seems like we shouldn't care if the timing was set, but rather whether there was already a main loop running? That is, I'd expect to see a bool that says "I have pushed a runtime keepalive", or "already running" or such?

Reply:
Yes that is exactly what this is supposed to mean.   I think after I investigated this I determined that a `has timing been set` was the same things as "is running".. but you are correct I can come up with a better name and some comments.